examples/resource-monitoring-app/resource-monitoring-common/src/instances/ActivatedCarbonFilterMonitoring.cpp

```diff
@@ -0,0 +1,48 @@
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/util/config.h>
+#include <instances/ActivatedCarbonFilterMonitoring.h>
+
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::HepaFilterMonitoring;
+using chip::Protocols::InteractionModel::Status;
+
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ResourceMonitoring;
+using chip::Protocols::InteractionModel::Status;
+
+//-- Hepa filter Monitoring Instance methods
+CHIP_ERROR ActivatedCarbonFilterMonitoringInstance::AppInit()
+{
+    ChipLogDetail(Zcl, "ActivatedCarbonFilterMonitoringDelegate::Init()");
+    return CHIP_NO_ERROR;
+}
+
+Status ActivatedCarbonFilterMonitoringInstance::OnResetCondition()
+{
+    ChipLogDetail(Zcl, "ActivatedCarbonFilterMonitoringDelegate::OnResetCondition()");
+
+    if (GetDegradationDirection() == DegradationDirectionEnum::kDown)
+    {
+        UpdateCondition(100);
+    }
+    else if (GetDegradationDirection() == DegradationDirectionEnum::kUp)
+    {
+        UpdateCondition(0);
+    }
+    UpdateChangeIndication(ChangeIndicationEnum::kOk);
+    if (emberAfContainsAttribute(0x1, Clusters::ActivatedCarbonFilterMonitoring::Id, Attributes::LastChangedTime::Id))
```

Comment:
Why 0x1?  Shouldn't this use the actual endpoint id we were given?

Reply:
It's application specific, and the app developer should know it's endpoints.
Nevertheless I added getters for the EndpointId and it's now checking for the correct one.